cs/test/core/GuidConversionTests.cs

```diff
@@ -0,0 +1,29 @@
+﻿namespace UnitTest
+{
+    using System;
+    using Bond;
+    using NUnit.Framework;
+
+    /// <summary>
+    /// Summary description for GuidConversionTests
+    /// </summary>
+    [TestFixture]
+    public class GuidConversionTests
+    {
+        [Test]
+        public void ImplicitConversion()
+        {
+            var systemGuid = Guid.NewGuid();
+            var bondGuid = (GUID)systemGuid;
+
+            // The Datas should be populated if it converted correctly
+            Assert.AreNotEqual(bondGuid.Data1, 0);

```

Comment:
static const int one = 1;
#define CPU_LITTLE_ENDIAN  (*(char*)(&one))

The CPU_LITTLE_ENDIAN macro will evaluate as True on a little-endian system, False on a Big-endian system.

Using Visual Studio, uint32_t _byteswap_long(uint32_t val) returns the byte-swapped value of val. For non-VS compilers, the needed code is easy to find. If the 32-bit part of the broken-out guid doesn’t match what it ought to, check to see if matches the byte-swapped form; if it does, emit a different failure message.

From: Christopher Warrington (MSFT) [mailto:notifications@github.com]
Sent: Thursday, April 7, 2016 5:17 PM
To: Microsoft/bond <bond@noreply.github.com>
Subject: Re: [Microsoft/bond] Add Implicit Conversion between Bond.GUID and System.Guid (#145)


In cs/test/core/GuidConversionTests.cs<https://github.com/Microsoft/bond/pull/145#discussion_r58965604>:

> +    using NUnit.Framework;

> +

> +    /// <summary>

> +    /// Summary description for GuidConversionTests

> +    /// </summary>

> +    [TestFixture]

> +    public class GuidConversionTests

> +    {

> +        [Test]

> +        public void ImplicitConversion()

> +        {

> +            var systemGuid = Guid.NewGuid();

> +            var bondGuid = (GUID)systemGuid;

> +

> +            // The Datas should be populated if it converted correctly

> +            Assert.AreNotEqual(bondGuid.Data1, 0);

I'd be more confident in the test if we used a static GUID and asserted the values of the Data1..Data4 members explicitly to make sure that each representation of the GUID is correct, instead of just that we can round trip a GUID.

(When you do this, you'll likely need to make an assumption about the endianess of the memory layout of the test machine. That's fine. Bonus points for encoding such an assumption in a test case.)

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly or view it on GitHub<https://github.com/Microsoft/bond/pull/145/files/61fe7e948a68b70f85d08a0e2859fc811777a78b#r58965604>